### PR TITLE
Problem status

### PR DESCRIPTION
**Pull request by geniusgordon**
problem status
visible
problem list

### CODE REVIEW

File: problem/templates/problem/problem_list.html [hunk 1]
```diff
@@ -12,20 +13,22 @@
...
-                  data-toggle="tooltip" data-placement="left" title="Rejudge"></span>
-          {% endif %}
-          &nbsp;{{p.pk}}
-        </td>
-
-        <td><a href="{% url 'problem:problem.views.detail' p.pk %}">{{p.pname}}</a>&nbsp;
-          {% if user == p.owner or user.is_admin %}
-            <a href="{% url 'problem:problem.views.edit' p.pk %}">
+            <button class="btn btn-info btn-xs">
+              <span id='rejudge' class='glyphicon glyphicon-refresh'
```
Comment (nickelcheng) [inline problem/templates/problem/problem_list.html]: Please hide rejudge button in this version.

File: problem/templates/problem/problem_list.html [hunk 2]
```diff
@@ -54,6 +52,9 @@
 $(document).ready(function () {
   $('[data-toggle="tooltip"]').tooltip();
   $('.progress').css('overflow', 'inherit');
+  $('#del_btn').click(function (){
```
Comment (nickelcheng) [inline problem/templates/problem/problem_list.html]: @Changron, can you use this to show your alert. I think this code is more light

File: problem/views.py [hunk 3]
```diff
@@ -69,6 +70,10 @@ def detail(request, pid):
     tag_form = TagForm()
     try:
         problem = Problem.objects.get(pk=pid)
+        last_contest = problem.contest_set.all().order_by('-start_time')
```
Comment (nickelcheng) [inline problem/views.py]: Did you implement ```contest_set``` yourself? Or it's provided by django?

File: problem/views.py [hunk 4]
```diff
@@ -69,6 +70,10 @@ def detail(request, pid):
     tag_form = TagForm()
     try:
         problem = Problem.objects.get(pk=pid)
```
Comment (nickelcheng) [inline problem/views.py]: If the problem is invisible and request user is now owenr or admin, raise permission deny

**Comment by nickelcheng** [comment]
Can you fix #126 as well?

**Comment by geniusgordon** [comment]
fixed

**Comment by nickelcheng** [comment]
Please hide rejudge button in this version.

**Comment by geniusgordon** [comment]
removed

**Comment by nickelcheng** [comment]
+1
@CHIAHSINHOU please test

**Comment by CHIAHSINHOU** [comment]
what should I do before I can use it ?

**Comment by geniusgordon** [comment]
fixed

File: problem/views.py [hunk 5]
```diff
@@ -69,6 +70,13 @@ def detail(request, pid):
...
+        if problem.visible and (user != problem.owner or not user.is_admin):
```
Comment (nickelcheng) [inline problem/views.py]: ``` if not problem.visible```

**Comment by nickelcheng** [comment]
Why does your PR become so large and contains so many commits (instead of merge commit)?

**Comment by bruce3557** [comment]
seems weird in commit log
please check again

**Pull request closed without merging**